disperser/common/v2/blobstore/dynamo_metadata_store.go

```diff
@@ -265,6 +265,25 @@ func (s *BlobMetadataStore) GetBlobMetadata(ctx context.Context, blobKey corev2.
 	return metadata, nil
 }
 
+// CheckBlobExists checks if a blob exists without fetching the entire metadata.
+func (s *BlobMetadataStore) CheckBlobExists(ctx context.Context, blobKey corev2.BlobKey) (bool, error) {
+	// Use GetItem with ProjectionExpression to minimize data transfer
+	item, err := s.dynamoDBClient.GetItem(ctx, s.tableName, map[string]types.AttributeValue{
```

Comment:
We can use projection as well for GetItem I think: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/dynamodb#GetItemInput

Reply:
Our current dynamodb client interface doesn't support projection but I've added it:)

Reply:
LG! Can we make it `GetItemWithInput` (similar to `QueryWithInput`) and passing in `GetItemInput` as param, so in the future if there are other needs to customize, it'll be easy?

Reply:
yes, makes sense! updated